flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/UnionITCase.scala

```diff
@@ -54,7 +54,22 @@ class UnionITCase(
   }
 
   @Test
-  def testTernaryUnion(): Unit = {
+  def testUnion(): Unit = {
```

Comment:
There were some concerns about the build time of the `flink-table` module lately. In the past, we added quite a few integration tests without thinking about the added coverage vs. build time trade-off.

I would propose to replace the `testUnionWithFilter()` and `testUnionWithJoin()` tests with your new methods because these methods do not add to test coverage, IMO.